src/components/EncounterBuilder/EncounterBuilder.jsx

```diff
@@ -1,15 +1,249 @@
 import PropTypes from 'prop-types';
 import './EncounterBuilder.css';
+import React, { useEffect, useState } from 'react';
+import { useApolloClient, useMutation, gql } from '@apollo/client';
+import {MonsterFilter} from '../MonsterFilter/MonsterFilter';
+
+const ENCOUNTER_BUILDER_MUTATION = gql`
+  mutation CreateEncounter(
+    $userName: String!, $encounterName: String!, $partySize: Int!, $partyLevel: Int!, $summary: String!, $description: String!, $treasure: String!, $encounterMonsterIndexes: [String!]!
+    ) {
+    createEncounter(input: {
+      userName: $userName,
+      encounterName: $encounterName,
+      partySize: $partySize,
+      partyLevel: $partyLevel,
+      summary: $summary,
+      description: $description,
+      treasure: $treasure,
+      encounterMonsterIndexes: $encounterMonsterIndexes
+    }) {
+      encounter {
+        userName
+        id
+        encounterName
+        partySize
+        partyLevel
+        summary
+        description
+        treasure
+        encounterMonsters {
+          monsterIndex
+          monsterName
+          }
+        }
+    errors
+  }
+}
+`;
+
+const EncounterBuilder = ({userName, setEncounterCreated}) => {
+  const [monsterList, setMonsterList] = useState([]);
+  const [monsters, setMonsters] = useState([]);
+  const [selectedSizeFilter, setSelectedSizeFilter] = useState('');
+  const [selectedNameFilter, setSelectedNameFilter] = useState('');
+  const [selectedArmorClassFilter, setSelectedArmorClassFilter] = useState('');
+  const [selectedHitPointsFilter, setSelectedHitPointsFilter] = useState(0);
+  const [filteredMonsters, setFilteredMonsters] = useState([]);
+  const [toShow, setToShow] = useState([]);
+  const [newEncounter, setNewEncounter] = useState({
+    userName: userName,
+    encounterName: '',
+    partySize: 0,
+    partyLevel: 0,
+    summary: '',
+    description: '',
+    treasure: '',
+    encounterMonsterIndexes: [],
+  })
+
+  const [encounterBuilder] = useMutation(ENCOUNTER_BUILDER_MUTATION, {
+    variables: {
+      userName: newEncounter.userName,
+      encounterName: newEncounter.encounterName,
+      partySize: newEncounter.partySize,
+      partyLevel: newEncounter.partyLevel,
+      summary: newEncounter.summary,
+      description: newEncounter.description,
+      treasure: newEncounter.treasure,
+      encounterMonsterIndexes: newEncounter.encounterMonsterIndexes,
+    },
+    onCompleted: (data) => {
+      const encounterName = data.createEncounter.encounter;
+      console.log(encounterName, "encounterCreated")
+      client.resetStore()
```

Comment:
Huzzah!!! So glad you were able to sus out what was causing the query to not fire off. Great work on figuring out that it was a caching issue, and finding a solution! ⭐

Reply:
[Resource I used if you're curious to learn more about it](https://www.apollographql.com/docs/react/caching/advanced-topics/)

Reply:
Working on the reset button now